source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,293 @@
+For Super Rentals, we want to be able to display a map showing where each rental is.  To implement this feature, we will take advantage of several Ember concepts:
+
+1. A component to display a map on each rental listing.
+2. A service to keep a cache of rendered maps to use in different places in the application.
+3. A utility function to create a map from the Google Maps API.
+
+We'll start by displaying the map and work our way back to using the Google Map API.
+
+
+### Display Maps With a Component
+
+We'll start by adding a component that shows the rental's city on a map.
+
+```app/templates/components/rental-listing.hbs{+14}
+<h2>{{rental.title}}</h2>
+<p>Owner: {{rental.owner}}</p>
+<p>Type: {{rental-property-type rental.type}} - {{rental.type}}</p>
+<p>Location: {{rental.city}}</p>
+<p>Number of bedrooms: {{rental.bedrooms}}</p>
+{{#if isImageShowing}}
+  <p>
+    <img alt={{rental.type}} src={{rental.image}} width="500"/>
+  </p>
+  <button {{action "imageHide"}}>Hide image</button>
+{{else}}
+  <button {{action "imageShow"}}>Show image</button>
+{{/if}}
+{{location-map location=rental.city}}
+```
+
+Next, generate the map component using Ember-CLI.
+
+```shell
+ember g component location-map
+```
+
+Running this command generates three files: a component JavaScript file, a template, and a test file.
+To help think through what we want our component to do, we'll implement a test case first.
+
+In this case we plan on having our Google Maps service handle map display.  
+Our component's job will be to take the results from the map service (which is a map element) and append it to an element in the component template.
+
+All we want to do in our component is make sure its appending the output of the maps service to an inner map container element.
+To limit the test to validating just this behavior we'll take advantage of the registration API to provide a mock maps service.
+In the mock service, define a method that will fetch the map based on location, called `getMapElement`.
+
+```tests/integration/components/location-map-test.js
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+let MockMapsService = Ember.Service.extend({
+  getMapElement(location) {
+    this.set('calledWithLocation', location);
+    // We create a div here to simulate our maps service,
+    // which will create and then cache the map element
+    return document.createElement('div');
+  }
+});
```

Comment:
Are we assuming folks are doing this with prior testing experience or not?  This is a great example of a mock service, but if we're assuming little to no knowledge of testing prior to this (which might be the case in cases of jQuery converts), we should probably spend a bit more time explaining what we're doing with a mock service ...  /cc @locks ?